src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -29,18 +29,18 @@ pub fn generate_lockfile(manifest_path: &Path, config: &Config)
 
 pub fn update_lockfile(manifest_path: &Path,
                        opts: &UpdateOptions) -> CargoResult<()> {
-    let package = try!(Package::for_path(manifest_path, opts.config));
-
-    let previous_resolve = match try!(ops::load_pkg_lockfile(&package,
-                                                             opts.config)) {
-        Some(resolve) => resolve,
-        None => bail!("a Cargo.lock must exist before it is updated")
-    };
 
     if opts.aggressive && opts.precise.is_some() {
         bail!("cannot specify both aggressive and precise simultaneously")
     }
 
+    let package = try!(Package::for_path(manifest_path, opts.config));
+    let path = Path::new(manifest_path);
+    if !path.parent().unwrap().join("Cargo.lock").exists(){
+       let _ = generate_lockfile(manifest_path, opts.config);
+    }
```

Comment:
It may be worth matching the output of the tests below, I think calling `generate_lockfile` in addition to `resolve_with_previous` will cause the registry to get updated twice, for example.

This may be best by just short circuiting this all entirely by just bypassing all the below logic to call one of the other lockfile-generating functions (basically `cargo update` without a lockfile is akin to `cargo generate-lockfile`)